sbc_translation/models/translate_connector.py

```diff
@@ -0,0 +1,159 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2014 Compassion CH (http://www.compassion.ch)
+#    Releasing children from poverty in Jesus' name
+#    @author: Emanuel Cino <ecino@compassion.ch>
+#
+#    The licence is in the file __openerp__.py
+#
+##############################################################################
+import datetime
+
+from openerp.addons.mysql_connector.model.mysql_connector \
+    import mysql_connector
+
+
+class TranslateConnect(mysql_connector):
+    """ Contains all the utility methods needed to talk with the MySQL server
+        used by translate platform, as well as all mappings
+        from OpenERP fields to corresponding MySQL fields.
+    """
+
+    def __init__(self):
+        super(TranslateConnect, self).__init__(
+            'mysql_translate_host',
+            'mysql_translate_user',
+            'mysql_translate_pw',
+            'mysql_translate_db')
+
+        self.current_time = datetime.datetime.now()
+
+    def upsert_text(self, sponsorship, letter):
+        """Push or update text (db table) on local translate platform
+        """
+        child = sponsorship.child_id
+        sponsor = sponsorship.partner_id
+
+        self.letter_name = "_".join((child.code, sponsor.ref, str(letter.id)))
+        child_age = datetime.date.today().year - int(child.birthdate[:4])
+
+        src_lang_id = self.get_language_id(
+            letter.original_language_id.code_iso)
+
+        # Define the target language
+        child_lang = child.project_id.country_id.spoken_lang_ids
+        # eng, fra, deu, spa, por, ita
+        translate_language_ids = [1, 2, 3, 4, 5, 14]

```

Comment:
You should never hardcode ids ! I would do :
`translate_language_ids = letter.env['res.lang.compassion'].search([('translatable', '=', True)]).ids`